pkg/util/volume.go

```diff
@@ -102,3 +103,29 @@ func getVol(ctx context.Context, client *k8sclient.K8sClient, pod *corev1.Pod, n
 	}
 	return
 }
+
+type VolumeLocks struct {
+	locks sync.Map
+	mux   sync.Mutex
+}
+
+func NewVolumeLocks() *VolumeLocks {
+	return &VolumeLocks{}
+}
+
+func (vl *VolumeLocks) TryAcquire(volumeID string) bool {
```

Comment:
should we introduce timeout for these type of locks? could they cause deadlock?

Reply:
this lock is not waiting, so it's unnecessary